quiz.js

```diff
@@ -1,168 +1,137 @@
         let data = await response.json();
-        //set questions equal to all returned data using the spread operator
         questions = [...data.results];
-        //call generateQuestion function
         generateQuestion();
-        console.log(questions)
     }
-    //this function generates new questions
-    function generateQuestion() {
 
-        //if statement to call endGame function once questions.length is reached
+    //this function generates the next question
+    function generateQuestion() {
+        //If we've gone through all the questions, end the game
         if (questionIndex === questions.length) {
             endGame();
         }
 
-        //emptys answer buttons
-        answerZoneEl.empty();
-        //emptys right/wrong area
-        rightWrongEl.empty();
-
-        cheer.pause();
-        jeer.pause();
-        function fix (errorString){
-            if(errorString.includes("&#039;")){
-                console.log("its in there")
-            }
-            else {(errorString.includes("&#039;"))
-                    console.log("its not there")
-            }
-            while(errorString.includes("&#039;")){
-                errorString = errorString.replace("&#039;", "\'")
-                }
-            while(errorString.includes("&quot;")){
-                errorString = errorString.replace("&quot;","\"")
-                }
-            while(errorString.includes("&amp;")){
-                errorString = errorString.replace("&amp;","&")
-            }
-            while(errorString.includes("&Uuml;")){
-                errorString = errorString.replace("&Uuml;","Ü")
-            }    
-            while(errorString.includes("&eacute;")){
-                errorString = errorString.replace("&eacute;","é")
-            }        
-            return errorString
-        }
-        //sets currentQuestion equal to questions index
+        resetQuestionZone();        
+        
         let currentQuestion = questions[questionIndex];
-        //set correctAnswer equal to the correct answer of the current question
-        correctAnswer = currentQuestion.correct_answer;
-        //tells user what question number they are on
+        correctAnswer       = fix(currentQuestion.correct_answer);
+
         questionNumberEl.text(`Question: ${questionIndex + 1} / 10`)
-        //tells user score
-        scoreEl.text(`Score: ${score}`);
-        //shows user the current question after a fix
-        let fixedCurrentQuestion = fix(currentQuestion.question)
-        questionEl.text(`Question: ${fixedCurrentQuestion}`);
-        //shows user what category the current question is from
-        categoryEl.text(`Category: ${currentQuestion.category}`);
-        //sets up array of all possible answer choices using the spread operator for incorrect answers
-        console.log(currentQuestion.incorrect_answers)
-        for (var i = 0; i < 2; i++) {
```

Comment:
Couple things wrong with this line. 
1) Don't hardcode `2` like this. `2` has no context on it. Instead, use `currentQuestion.incorrect_answers.length`. If the api changed to not serve 3 incorrect answers, this would need reworked. 
2) From what the api serves, it should either be `i <= 2` or `i < 3`. Currently, it'll only call `fix` on the answers at indexes 0 and 1, but not 2.